docs/working-copy.md

```diff
@@ -12,12 +12,16 @@ working-copy contents when they have changed. Most `jj` commands you run will
 commit the working-copy changes if they have changed. The resulting revision
 will replace the previous working-copy revision.
 
-Also unlike most other VCSs, added files are implicitly tracked. That means that
-if you add a new file to the working copy, it will be automatically committed
-once you run e.g. `jj st`. Similarly, if you remove a file from the working
-copy, it will implicitly be untracked. To untrack a file while keeping it in
-the working copy, first make sure it's [ignored](#ignored-files) and then run
-`jj file untrack <path>`.
+Also unlike most other VCSs, added files are implicitly tracked by default. That
+means that if you add a new file to the working copy, it will be automatically
+committed once you run e.g. `jj st`. Similarly, if you remove a file from the
+working copy, it will implicitly be untracked. The `snapshot.auto-track` config
+option controls which paths get automatically tracked when they're added. See
+the [fileset documentation](filesets.md) for the syntax.
+
+You can use `jj file untrack` to untrack a file while keeping it in the working
+copy. However, first [ignore](#ignored-files) them or remove them from the
+`snapshot.auto-track` patterns; otherwise they will be immediately tracked again.
```

Comment:
Shouldn't `track` and `untrack` bring up the ignore file or the config file in the editor, so it is more of an atomic action?

Reply:
My current opinion is that something like this can probably be made to work and is a good suggestion in general, but there are enough devils sitting in the details that we might not want to implement that right now and right here.

So, I'd put this discussion outside the scope of this PR.

---------

That said, I thought a bit about either putting the list of ignored files in an exclude file or putting it into a config file, and here is a proposal for the latter (which seemed somewhat easier to me).

If we put the list of (un)tracked files in a TOML config file, the main questions in my mind are:
- Should `jj file (un)track` modify the repo-level config file or workspace-level, like https://github.com/martinvonz/jj/pull/3670? If it's the latter, we should implement it.
- Do we attempt to sync this config with Git's `.git/info/excludes` file? (Fortunately, I think we can always just say no to this)
- How do we deal with the possible situation if these configs are specified manually (as opposed to with `jj file track` which wouldn't do this) in both the repo-level and the user-level config files (and perhaps also in the command-line args or workspace-level). 
 
For the last question, I believe we currently have the more specific config fully override the less specific config. In particular, if both are TOML lists, they are *not* concatenated, the latter replaces the former. We could change that, but then it would be unclear how and whether to make it possible to remove an item from the list in the more specific config.

One trick that comes to mind is this. Have the config have the form:

```toml
[ignore_patterns]
global_list1=["pattern1", "pattern2", ]
global_list2=["pattern3"]
```

The names here are arbitrary. If that is the repo-level config, the workspace-level config could add a new list by adding `ignored_patterns.local_list1=["pattern5"]`. Or, it could adjust the top-level config by redefining `global_list1`.

Reply:
> My current opinion is that something like this can probably be made to work and is a good suggestion in general, but there are enough devils sitting in the details that we might not want to implement that right now and right here.

I agree with this but think that we also should consider this as a potential solution for our own `.ignore` in #3525. Because it just eliminates one state.

Reply:
Closing this